hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/clientrm/RouterYarnClientUtils.java

```diff
@@ -468,5 +474,65 @@ public static GetResourceProfileResponse mergeClusterResourceProfileResponse(
     profileResponse.setResource(resource);
     return profileResponse;
   }
+
+  /**
+   * Merges a list of GetAttributesToNodesResponse.
+   *
+   * @param responses a list of GetAttributesToNodesResponse to merge.
+   * @return the merged GetAttributesToNodesResponse.
+   */
+  public static GetAttributesToNodesResponse mergeAttributesToNodesResponse(
+      Collection<GetAttributesToNodesResponse> responses) {
+    GetAttributesToNodesResponse attributesToNodesResponse =
+        GetAttributesToNodesResponse.newInstance(new HashMap<>());
+    Map<NodeAttributeKey, List<NodeToAttributeValue>> nodeAttributeMap = new HashMap<>();
+    for (GetAttributesToNodesResponse response : responses) {
+      if (response != null && response.getAttributesToNodes() != null) {
+        nodeAttributeMap.putAll(response.getAttributesToNodes());
+      }
+    }
+    attributesToNodesResponse.setAttributeToNodes(nodeAttributeMap);
+    return attributesToNodesResponse;
```

Comment:
Can we do something like:
```
return GetAttributesToNodesResponse.newInstance(nodeAttributeMap);
```

Reply:
Thanks for your suggestion, I will modify the code.